firewall/interception/windowskext/kext.go

```diff
@@ -25,108 +25,57 @@ var (
+// Init initializes the DLL and the Kext (Kernel Driver).
+func Init(path string) error {
+	driverPath = path
+	return nil
 }
 
-// Init initializes the DLL and the Kext (Kernel Driver).
-func Init(dllPath, driverPath string) error {
+// Start intercepting.
+func Start() error {
+	kextLock.Lock()
+	defer kextLock.Unlock()
 
-	new := &WinKext{
-		driverPath: driverPath,
-	}
+	filename := `\\.\` + driverName
 
+	// check if driver is already installed
 	var err error
-
-	// load dll
-	new.dll, err = windows.LoadDLL(dllPath)
-	if err != nil {
-		return err
+	kextHandle, err = openDriver(filename)
+	if err == nil {
+		return nil // device was already initialized
 	}
 
-	// load functions
-	new.init, err = new.dll.FindProc("PortmasterInit")
-	if err != nil {
-		return fmt.Errorf("could not find proc PortmasterStart in dll: %s", err)
-	}
-	new.start, err = new.dll.FindProc("PortmasterStart")
+	// initialize and start driver service
+	service, err := driverInstall(driverPath)
 	if err != nil {
-		return fmt.Errorf("could not find proc PortmasterStart in dll: %s", err)
+		return fmt.Errorf("Failed to start service: %s", err)
 	}
-	new.stop, err = new.dll.FindProc("PortmasterStop")
-	if err != nil {
-		return fmt.Errorf("could not find proc PortmasterStop in dll: %s", err)
-	}
-	new.recvVerdictRequest, err = new.dll.FindProc("PortmasterRecvVerdictRequest")
-	if err != nil {
-		return fmt.Errorf("could not find proc PortmasterRecvVerdictRequest in dll: %s", err)
-	}
-	new.setVerdict, err = new.dll.FindProc("PortmasterSetVerdict")
-	if err != nil {
-		return fmt.Errorf("could not find proc PortmasterSetVerdict in dll: %s", err)
-	}
-	new.getPayload, err = new.dll.FindProc("PortmasterGetPayload")
-	if err != nil {
-		return fmt.Errorf("could not find proc PortmasterGetPayload in dll: %s", err)
-	}
-	new.clearCache, err = new.dll.FindProc("PortmasterClearCache")
-	if err != nil {
-		// the loaded dll is an old version
-		log.Errorf("could not find proc PortmasterClearCache (v1.0.12+) in dll: %s", err)
-	}
-
-	// initialize dll/kext
-	rc, _, lastErr := new.init.Call()
-	if rc != windows.NO_ERROR {
-		return formatErr(lastErr, rc)
-	}
-
-	// set kext
-	kextLock.Lock()
-	defer kextLock.Unlock()
-	kext = new
-
-	return nil
-}
 
-// Start intercepting.
-func Start() error {
-	kextLock.Lock()
-	defer kextLock.Unlock()
+	// open the driver
+	kextHandle, err = openDriver(filename)
 
-	// convert to C string
-	charArray := make([]byte, len(kext.driverPath)+1)
-	copy(charArray, []byte(kext.driverPath))
-	charArray[len(charArray)-1] = 0 // force NULL byte at the end
+	// close the service handles
+	_ = windows.DeleteService(service)
+	_ = windows.CloseServiceHandle(service)
 
-	rc, _, lastErr := kext.start.Call(
-		uintptr(unsafe.Pointer(&charArray[0])),
-	)
-	if rc != windows.NO_ERROR {
-		return formatErr(lastErr, rc)
+	// driver was not installed
+	if err != nil {
+		return fmt.Errorf("Failed to start the kext service: %s %q", err, filename)
```

Comment:
Error messages should always start lower case.
Also, please use `%w` to wrap the error.
I would include the filename in the message and put the error last, as it may be long. This will make it easier to make sense of the message.